applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationAsyncClient.java

```diff
@@ -27,6 +27,73 @@
  * Client that contains all the operations for {@link ConfigurationSetting ConfigurationSettings} in Azure Configuration
  * Store.
  *
+ * <p>
+ *     Azure Configuration Store operations allowed by the client are adding, retrieving, updating, and deleting
+ *     ConfigurationSettings, additionally more complex operations of listing settings or revision of a setting based
+ *     on a filter are supported as well.
+ * </p>
+ *
+ * Construct the client
+ * <pre>
+ *     {@code
+ *     String connectionString = <connection_string>;
+ *     ConfigurationClientCredentials credentials = new ConfigurationClientCredentials(connectionString);
+ *     ConfigurationAsyncClient client = ConfigurationAsyncClient.builder()
+ *          .credentials(credentials)
+ *          .build();
+ *     }
+ * </pre>
+ *
+ * Add a Configuration Setting
+ * <pre>
+ *     {@code
+ *     client.addSetting("prodDBConnection", "<db_connection>").subscribe(result -> {
+ *         ConfigurationSetting setting = result.value();
+ *         System.out.println(String.format("Key: %s, Value: %s", setting.key(), setting.value());
+ *     });
+ *
+ *     client.setSetting("testDBConnection", "<db_connection>").subscribe(result -> {
+ *         ConfigurationSetting setting = result.value();
+ *         System.out.println(String.format("Key: %s, Value: %s", setting.key(), setting.value());
+ *     });
+ *     }
+ * </pre>
+ *
+ * Retrieve a Configuration Setting
+ * <pre>
+ *     {@code
+ *     client.getSetting("testDBConnection").subscribe(result -> {
+ *         ConfigurationSetting setting = result.value();
+ *         System.out.println(String.format("Key: %s, Value: %s", setting.key(), setting.value());
+ *     });
+ *     }
+ * </pre>
+ *
+ * Update an existing Configuration Setting
+ * <pre>
+ *     {@code
+ *     client.updateSetting("prodDBConnection", "<updated_db_connection").subscribe(result -> {
+ *         ConfigurationSetting setting = result.value();
+ *         System.out.println(String.format("Key: %s, Value: %s", setting.key(), setting.value());
+ *     });
+ *
+ *     client.setSetting("testDBConnection", "<updated_db_connection>").subscribe(result -> {
```

Comment:
Might be worthwhile to describe that `setSetting` can add or update a configuration setting. From the heading "Update an existing Configuration Setting", it seems like you need to perform both an `updateSetting` followed by a `setSetting`.